core/client/src/main/java/alluxio/client/netty/NettyClient.java

```diff
@@ -83,4 +95,26 @@ public void initChannel(SocketChannel ch) throws Exception {
 
     return boot;
   }
+
+  /**
+   * Note: Packet streaming requires {@link io.netty.channel.epoll.EpollMode} to be set to
+   * LEVEL_TRIGGERED which is not supported in netty versions < 4.0.26.Final. Without shading
+   * netty in Alluxio, we cannot use epoll.
+   *
+   * @return {@link ChannelType} to use
+   */
+  private static ChannelType getChannelType() {
+    boolean epollModeSupported = true;
+    try {
+      EpollChannelConfig.class.getField("EPOLL_MODE");
+    } catch (NoSuchFieldException e) {
+      LOG.warn("EPOLL_MODE is not supported in netty with version < 4.0.26.Final.");
+      epollModeSupported = false;
+    }
+
+    if (PACKET_STREAMING_ENABLED && !epollModeSupported) {
```

Comment:
Could we put the epoll detection logic in an if clause since it is only relevant when packet streaming is enabled?